src/net_processing.cpp

```diff
@@ -2470,6 +2470,19 @@ unsigned int ProcessMessages(CNode* pfrom, CConnman& connman, std::atomic<bool>&
             LOCK(pfrom->cs_vProcessMsg);
             if (pfrom->vProcessMsg.empty())
                 return 0;
+
+            // Check if we are supposed to avoid cs_main, but are about to try to lock it
```

Comment:
cs_main is taken by ProcessGetData  at like 2456 above. (github won't let me comment there).